src/frontend/src/tests/btc/components/fee/UtxosFeeContext.spec.ts

```diff
@@ -0,0 +1,150 @@
+import UtxosFeeContext from '$btc/components/fee/UtxosFeeContext.svelte';
+import * as btcSendApi from '$btc/services/btc-send.services';
+import {
+	initUtxosFeeStore,
+	UTXOS_FEE_CONTEXT_KEY,
+	type UtxosFeeStore
+} from '$btc/stores/utxos-fee.store';
+import { BTC_MAINNET_NETWORK_ID, ICP_NETWORK_ID } from '$env/networks.env';
+import * as authStore from '$lib/derived/auth.derived';
+import * as authServices from '$lib/services/auth.services';
+import { mockUtxosFee } from '$tests/mocks/btc.mock';
+import { mockIdentity } from '$tests/mocks/identity.mock';
+import { mockPage } from '$tests/mocks/page.store.mock';
+import type { Identity } from '@dfinity/agent';
+import { render, waitFor } from '@testing-library/svelte';
+import { readable } from 'svelte/store';
+import { expect } from 'vitest';
+
+describe('UtxosFeeContext', () => {
+	const amount = 10;
+	const networkId = BTC_MAINNET_NETWORK_ID;
+	const mockContext = (store: UtxosFeeStore) => new Map([[UTXOS_FEE_CONTEXT_KEY, { store }]]);
+	const mockBtcSendApi = () =>
+		vi.spyOn(btcSendApi, 'selectUtxosFee').mockResolvedValue(mockUtxosFee);
+	const mockAuthStore = (value: Identity | null) =>
+		vi.spyOn(authStore, 'authIdentity', 'get').mockImplementation(() => readable(value));
+
+	const props = {
+		amount,
+		networkId
+	};
+
+	beforeEach(() => {
+		mockPage.reset();
+	});
+
+	it('should call selectUtxosFee with proper params', async () => {
+		const store = initUtxosFeeStore();
+		const setUtxosFeeSpy = vi.spyOn(store, 'setUtxosFee');
+		const selectUtxosFeeSpy = mockBtcSendApi();
+
+		mockAuthStore(mockIdentity);
+
+		render(UtxosFeeContext, {
+			props,
+			context: mockContext(store)
+		});
+
+		await waitFor(() => {
+			expect(selectUtxosFeeSpy).toHaveBeenCalledOnce();
+			expect(selectUtxosFeeSpy).toBeCalledWith({
+				amount,
+				network: 'mainnet',
+				identity: mockIdentity
+			});
+			expect(setUtxosFeeSpy).toHaveBeenCalledOnce();
+			expect(setUtxosFeeSpy).toHaveBeenCalledWith({ utxosFee: mockUtxosFee });
+		});
+	});
+
+	it('should not call selectUtxosFee if no authIdentity available', async () => {
+		const store = initUtxosFeeStore();
+		const selectUtxosFeeSpy = vi.spyOn(authServices, 'nullishSignOut').mockResolvedValue();
+
+		mockAuthStore(null);
+
+		render(UtxosFeeContext, {
+			props,
+			context: mockContext(store)
+		});
+
+		await waitFor(() => {
+			expect(selectUtxosFeeSpy).toHaveBeenCalledOnce();
+		});
+	});
+
+	it('should not call selectUtxosFee if no networkId provided', async () => {
+		const store = initUtxosFeeStore();
+		const resetSpy = vi.spyOn(store, 'reset');
+		const selectUtxosFeeSpy = mockBtcSendApi();
+		const { networkId: _, ...newProps } = props;
+
+		mockAuthStore(mockIdentity);
```

Comment:
To avoid duplicating `mockAuthStore(mockIdentity);` you can have two distinct describe with a before each

```
describe("not signed-in", () =>
    beforeEach
       mockAuthStore(undefined);

describe("signed-in", () =>
   beforeEach
      mockAuthStore(mockIdentity);
```

Just sharing if interesting, not a call to action.

Reply:
I kept the initial structure but shortened it a bit by using `mockIdentity` as default value and passing `null` only in the related test.